sklearn/tree/_criterion.pyx

```diff
@@ -667,20 +714,34 @@ cdef class RegressionCriterion(Criterion):
         self.weighted_n_samples = weighted_n_samples
         self.weighted_n_node_samples = 0.
```

Comment:
Shouldn't this be defined inside `pointer_reset`?

Reply:
Thank you! I'll make sure variables elsewhere are where they supposed to be